pkg/jobstore/boltdb/context_test.go

```diff
@@ -0,0 +1,97 @@
+//go:build unit || !integration
```

Comment:
Is it worth testing the case where a context contains more than a single transaction? Or will that never happen?

Reply:
boltdb doesn't allow concurrent active transactions. If you attempt to begin a transaction while another one is still active, whether in the same context or note, it will block. I've added additional tests in https://github.com/bacalhau-project/bacalhau/pull/3996/commits/b0c7060a3e822fbf2cc59eabfaa42fa286882c4d for botldb jobstore to verify we are handling this gracefully

Reply:
I believe boltdb supports concurrent read-only transactions: https://github.com/boltdb/bolt?tab=readme-ov-file#transactions